### PR TITLE
Enable maven checkstyle plugins by default for all new pinot components and fix pinot-spi styling

### PR DESCRIPTION
**Pull request by xiangfu0**
## Description
- Enable maven checkstyle plugin by default with warning severity. All new pinot components should honor this code styling.
- Fix checkstyle.xml with checkstyle lib upgrade
- Fix `pinot-spi` module code styling

Right now we have violations on almost all the components, so I put the prop: `checkstyle.fail.on.violation` to all the component pom files to suppress the errors.

Then we can fix the violations and remove those props one by one. Tracking issue: https://github.com/apache/pinot/issues/7309

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/InstanceReplicaGroupPartitionConfig.java [hunk 1]
```diff
@@ -41,7 +41,8 @@
   @JsonPropertyDescription("Number of partitions for replica-group based selection, do not partition the replica-group (1 partition) if not specified")
   private final int _numPartitions;
 
-  @JsonPropertyDescription("Number of instances per partition (within a replica-group) for replica-group based selection, select all instances if not specified")
+  @JsonPropertyDescription(
+      "Number of instances per partition (within a replica-group) for replica-group based selection, select all instances if not specified")
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/InstanceReplicaGroupPartitionConfig.java:45]: This line is still too long?

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/InstanceAssignmentConfig.java [hunk 2]
```diff
@@ -31,10 +31,12 @@
...
   private final InstanceTagPoolConfig _tagPoolConfig;
 
-  @JsonPropertyDescription("Configuration for the instance constraints of the instance assignment, which filters out unqualified instances and sorts instances for picking priority")
+  @JsonPropertyDescription("Configuration for the instance constraints of the instance assignment,"
+      + " which filters out unqualified instances and sorts instances for picking priority")
   private final InstanceConstraintConfig _constraintConfig;
 
-  @JsonPropertyDescription("Configuration for the instance replica-group and partition of the instance assignment (mandatory)")
+  @JsonPropertyDescription(
+      "Configuration for the instance replica-group and partition of" + " the instance assignment (mandatory)")
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/InstanceAssignmentConfig.java]: This looks slightly weird. One way to fix is to add empty comment `//` to the first line to force it to change the line

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java [hunk 3]
```diff
@@ -479,6 +479,7 @@ public void validate() {
             default:
               throw new IllegalStateException("Unsupported data type: " + dataType + " in COMPLEX field: " + fieldName);
           }
+          break;
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java:482]: +1

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java [hunk 4]
```diff
@@ -804,11 +804,12 @@ private static String constructTransformFunctionString(String incomingName, int
           innerFunction = String.format("fromEpochDays(%s)", incomingName);
         }
         break;
+      default:
+        throw new RuntimeException("Unsupported incomingTimeUnit - " + incomingTimeUnit);
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java:808]: We usually throw `IllegalStateException` here

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java [hunk 5]
```diff
@@ -839,6 +840,8 @@ private static String constructTransformFunctionString(String incomingName, int
           outerFunction = String.format("toEpochDays(%s)", innerFunction);
         }
         break;
+      default:
+        throw new RuntimeException("Unsupported outgoingTimeUnit - " + outgoingTimeUnit);
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java:844]: Same

File: pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java [hunk 6]
```diff
@@ -92,17 +92,17 @@ public PinotConfiguration() {
   /**
    * Builds a new instance out of an existing Apache Commons {@link Configuration} instance. Will apply property key sanitization to enable relaxed binding.
    * Properties from the base configuration will be copied and won't be linked.
-   * 
+   *
    * @param baseConfiguration an existing {@link Configuration} for which all properties will be duplicated and sanitized for relaxed binding. 
    */
   public PinotConfiguration(Configuration baseConfiguration) {
-    this.configuration =
+    this._configuration =
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java]: Remove `this.`, same for other places

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/ConfigUtils.java [hunk 7]
```diff
@@ -31,6 +31,9 @@
 public class ConfigUtils {
   private static final Map<String, String> ENVIRONMENT_VARIABLES = System.getenv();
 
+  private ConfigUtils() {
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/config/ConfigUtils.java:34]: (nit) We usually put this right next to the class definition for slightly better readability, see other utils class for example. Same for other util classes

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/InstanceReplicaGroupPartitionConfig.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/InstanceReplicaGroupPartitionConfig.java:45]: Although the error msg shows don't go beyond 120, the actual limit is 160 :p

**Pull request merged**